src/fft.jl

```diff
@@ -121,20 +119,22 @@ function compute_final_fft_size(model::Model{T}, fft_size, Ecut, supersampling,
             fft_size = compute_fft_size_precise(model.lattice, Ecut, kpoints_temp;
                                                 supersampling=supersampling)
         end
+    end
 
-        # Sanity checks
+    # validate
+    if variational
         max_E = sum(abs2, model.recip_lattice * floor.(Int, Vec3(fft_size) ./ 2)) / 2
         Ecut > max_E && @warn(
             "For a variational method, Ecut should be less than the maximal kinetic " *
             "energy the grid supports ($max_E)"
         )
     else
-        # ensure fft_size is provided, and other options are not set
-        # TODO make proper error messages when the interface gets a bit cleaned up
+        # ensure no other options are set
         @assert fft_size !== nothing
```

Comment:
This assert can never be false and could actually be misleading with the comment line you put above it.

Reply:
Indeed, good catch